analyses/cell-type-wilms-tumor-06/notebook_template/06_cnv_infercnv_exploration.Rmd

```diff
@@ -0,0 +1,445 @@
+---
+
+```{r setup, include=FALSE}
+
+knitr::opts_chunk$set(echo = TRUE,
+                      message=FALSE,
+                      warnings=FALSE
+                      )
+
+```
+
+
+## Introduction
+
+
+```{r}
+
+subdiagnosis <- readr::read_tsv(
+  file.path("..", "..", "..", "data", "current", params$scpca_project_id, "single_cell_metadata.tsv"),
+  show_col_types = FALSE
+  ) |>
+  dplyr::filter(scpca_sample_id == params$sample_id) |>
+  dplyr::pull(subdiagnosis)
+
+```
+
+This notebook explores using [`infercnv`](https://github.com/broadinstitute/inferCNV/wiki) to estimate tumor and normal cells in `r params$sample_id` from SCPCP000006. 
+This sample has a(n) `r subdiagnosis` subdiagnosis.
+
+`infercnv` was run using the `06_inferCNV.R` script with and without a normal reference. We also tested the impact of the subselection of normal cells using either immune, and/or endothelial cells as healthy reference.
+
+In this notebook, we just want to compare the heatmaps of CNV profiles, and evaluate how comparable are the methods and how sensible they are to key parameters such as selection of healthy reference. 
+
+
+### Libraries
+
+
+```{r}
+
+library(infercnv)
+library(SCpubr)
+library(ggplot2)
+library(patchwork)
+library(Seurat)
+
+```
+
+
+### Functions
+
+Here we defined function that will be used multiple time all along the notebook. 
+
+#### Visualize CNV grouped by clusters or other metadata
+
+For a Seurat object `object`, the function `Do_CNV_heatmap` load the `infercnv` object created with the script `06_infercnv.R` using `reference_value` as a reference and call the function `SCpubr::do_CopyNumberVariantPlot` to plot the mean CNV score in each group defined in `group.by`.
+
+- `object` is the Seurat object 
+
+- `infercnv_obj` is the `infercnv` object
+
+- `group.by` is the metadata used for grouping the violin plots
+
+```{r}
+
+Do_CNV_heatmap <- function(object, infercnv_obj, group.by){
+  
+
+  chromosome_locations = SCpubr::human_chr_locations
+
+  out <- SCpubr::do_CopyNumberVariantPlot(sample = object,
+                                        infercnv_object = infercnv_obj,
+                                        using_metacells = FALSE,
+                                        chromosome_locations = chromosome_locations,
+                                        return_object = FALSE,
+                                        group.by = group.by
+                                        )
+  out <- out + 
+    ggtitle(glue::glue("Copy Number Variant Plot, ", reference_value)) +
+    ylab(label = "")
+  
+  return(out)
+}
+
+```
+
+#### Calculate a global CNV score per cell to check the general distribution
+
+For a `Seurat` object an `infercnv` object created with the script `06_infercnv.R` using `reference_value` as a reference, the function `Do_CNV_score` load the `infercnv` object and calculate a CNV score per cell.
+The score is calculated based on the [biostar discussion](https://www.biostars.org/p/9573777/).
```

Comment:
Is there a particular reason why you chose boundaries of 0.9 and 1.1, since 0.95 and 1.05 is presented in the discussion? It's worth explaining that reasoning here

Reply:
hm right, actually I first used 0.95 and 1.05 and the results were not really convincing (no higher scores for normal cells) so I decided to be more strict with the thresholds. But as you see it hasn't improved... not sure what to do with this score at that stage, might be to remove. But in the exploratory notebook might also be to keep and explain that it is not what we would have expect.